nucliadb_texts/src/reader.rs

```diff
@@ -88,9 +90,48 @@ impl Debug for TextReaderService {
 
 impl FieldReader for TextReaderService {
     #[tracing::instrument(skip_all)]
-    fn pre_filter(&self, _request: &PreFilterRequest) -> NodeResult<PreFilterResponse> {
-        // Right now it returns an empty response
-        Ok(PreFilterResponse::default())
+    fn pre_filter(&self, request: &PreFilterRequest) -> NodeResult<PreFilterResponse> {
+        let mut timestamp_queries = Vec::new();
+        for timestamp_query in request.timestamp_filters.iter() {
+            let from = timestamp_query.from.as_ref();
+            let to = timestamp_query.to.as_ref();
+            let field = match timestamp_query.applies_to {
+                FieldDateType::Created => self.schema.created,
+                FieldDateType::Modified => self.schema.modified,
+            };
+            if let Some(query) = search_query::produce_date_range_query(field, from, to) {
+                let query: Box<dyn Query> = Box::new(query);
+                timestamp_queries.push((Occur::Should, query));
+            }
+        }
+        let pre_filter_query = BooleanQuery::new(timestamp_queries);
+        let searcher = self.reader.searcher();
+        let docs_fulfilled = searcher.search(&pre_filter_query, &DocSetCollector)?;
+        let mut valid_fields = Vec::new();
+        for fulfilled_doc in docs_fulfilled {
+            let Ok(fulfilled_doc) = searcher.doc(fulfilled_doc) else {
+                error!("Could not retrieve {fulfilled_doc:?}");
+                continue;
+            };
+            let resource_id = fulfilled_doc
+                .get_first(self.schema.uuid)
+                .expect("documents must have a uuid.")
```

Comment:
Same here, isn't expect too hard?

Reply:
If you look at search, we are unwrapping in the same situation. Either way I do think it should be an panic, if we are not indexing that data, which is part of our schema, something is very wrong.